source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/ProcessTypeValidator.cs

```diff
@@ -12,17 +12,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+using Energinet.DataHub.EDI.IncomingMessages.Domain.Messages;
 using Energinet.DataHub.EDI.IncomingMessages.Infrastructure.ValidationErrors;
 
 namespace Energinet.DataHub.EDI.IncomingMessages.Infrastructure.Messages.RequestAggregatedMeasureData;
 
 public class ProcessTypeValidator : IProcessTypeValidator
 {
-    private static readonly IReadOnlyCollection<string> _whiteList = new[] { "D03", "D04", "D05", "D32" };
+    private static readonly IReadOnlyCollection<string> _aggregatedMeasureDataWhitelist =
+        new[] { "D03", "D04", "D05", "D32" };
 
-    public async Task<Result> ValidateAsync(string processType, CancellationToken cancellationToken)
+    private static readonly IReadOnlyCollection<string> _wholesaleServicesWhitelist =
+        new[] { "D05", "D32" };
+
+    public async Task<Result> ValidateAsync(IIncomingMessage message, CancellationToken cancellationToken)
     {
-        return await Task.FromResult(_whiteList.Contains(processType) ?
-             Result.Succeeded() : Result.Failure(new NotSupportedProcessType(processType))).ConfigureAwait(false);
+        return await Task.FromResult(
+                message switch
+                {
+                    RequestAggregatedMeasureDataMessage ragdm =>
```

Comment:
```suggestion
                    RequestAggregatedMeasureDataMessage ramdm =>
```
That's the naming in "source/IncomingMessages.Infrastructure/Messages/RequestAggregatedMeasureData/MessageTypeValidator.cs"